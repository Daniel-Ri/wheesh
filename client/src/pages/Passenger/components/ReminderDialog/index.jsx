import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { Button, Dialog } from '@mui/material';

import classes from './style.module.scss';

const ReminderDialog = ({ open, handleClose, handleDelete }) => (
  <Dialog data-testid="ReminderDialog" open={open} onClose={handleClose}>
    <div className={classes.dialog}>
      <div className={classes.content}>
        <h2>
          <FormattedMessage id="app_verification" />
        </h2>
        <div className={classes.message}>
          <FormattedMessage id="app_sure_deleting_the_passenger" />
        </div>
      </div>
      <div className={classes.buttons}>
        <Button data-testid="CancelButton" variant="outlined" className={classes.cancel} onClick={handleClose}>
          <FormattedMessage id="app_cancel" />
        </Button>
        <Button data-testid="ConfirmButton" variant="contained" className={classes.confirm} onClick={handleDelete}>
          <FormattedMessage id="app_confirm" />
        </Button>
      </div>
    </div>
  </Dialog>
);

ReminderDialog.propTypes = {
  open: PropTypes.bool.isRequired,
  handleClose: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired,
};

export default ReminderDialog;
