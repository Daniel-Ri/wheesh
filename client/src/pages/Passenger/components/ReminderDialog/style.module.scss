@import '@styles/base';

.dialog {
  width: 20rem;
  padding: 1rem;
  display: flex;
  flex-direction: column;
  justify-content: space-between;

  .content {
    display: flex;
    flex-direction: column;
    gap: 1rem;

    h2 {
      text-align: center;
      color: $color-marine-blue;
      font-size: 120%;
    }
  }

  .buttons {
    margin-top: 1rem;
    margin-bottom: 0.25rem;
    display: flex;
    justify-content: flex-end;
    gap: 1rem;

    .cancel {
      text-transform: capitalize;
    }

    .confirm {
      text-transform: capitalize;
      background-color: $color-strawberry-red;
    }
  }
}