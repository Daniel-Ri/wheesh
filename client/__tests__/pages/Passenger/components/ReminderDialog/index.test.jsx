import { render, fireEvent } from '@utils/testHelper';
import ReminderDialog from '@pages/Passenger/components/ReminderDialog';

const mockHandleClickClose = jest.fn();
const mockHandleClickDelete = jest.fn();
let wrapper;

const mockProps = {
  open: true,
  handleClose: mockHandleClickClose,
  handleDelete: mockHandleClickDelete,
};

beforeEach(() => {
  wrapper = render(<ReminderDialog {...mockProps} />);
});

describe('Component ReminderDialog of Passenger page', () => {
  test('Renders ReminderDialog component', () => {
    const { getByTestId } = wrapper;
    expect(getByTestId('ReminderDialog')).toBeInTheDocument();
  });

  test('Calls handleClickClose when button is clicked', () => {
    const { getByTestId } = wrapper;

    const cancelBtn = getByTestId('CancelButton');
    fireEvent.click(cancelBtn);

    expect(mockHandleClickClose).toHaveBeenCalledTimes(1);
  });

  test('Calls handleClickDelete when button is clicked', () => {
    const { getByTestId } = wrapper;

    const confirmBtn = getByTestId('ConfirmButton');
    fireEvent.click(confirmBtn);

    expect(mockHandleClickDelete).toHaveBeenCalledTimes(1);
  });

  test('Should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
