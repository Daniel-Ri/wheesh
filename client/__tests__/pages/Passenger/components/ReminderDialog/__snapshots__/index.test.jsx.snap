// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ReminderDialog of Passenger page Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
      data-testid="ReminderDialog"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
        role="presentation"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby=":r3:"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
          role="dialog"
        >
          <div
            class="dialog"
          >
            <div
              class="content"
            >
              <h2>
                Verifikasi
              </h2>
              <div
                class="message"
              >
                Apakah Anda yakin untuk menghapus penumpang ini?
              </div>
            </div>
            <div
              class="buttons"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium cancel css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
                data-testid="CancelButton"
                tabindex="0"
                type="button"
              >
                Batal
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium confirm css-sghohy-MuiButtonBase-root-MuiButton-root"
                data-testid="ConfirmButton"
                tabindex="0"
                type="button"
              >
                Konfirmasi
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
